core/templates/dev/head/services/IdGenerationService.js

```diff
@@ -19,7 +19,7 @@
 oppia.factory('IdGenerationService', [function() {
   return {
     generateNewId: function() {
-      return Math.random().toString(36).slice(2);
+      return Math.random().toString(36).slice(2).substring(0, 10);
```

Comment:
@lilithxxx, this looks good! can you add a comment above line 22 explaining why we only use last 10 digits. Also, add a test for this i.e, `should generate Id of fixed length`.
After that this PR will be good to go and thanks for taking look into this.

Reply:
Note: I suggest taking the last 10 digits of the number (better entropy) rather than first 10 digits.

Reply:
@DubeySandeep yes sure just one thing. The test that's added(generate a random id) also tests for its length. Would it be wise to add another test just for its length?

Reply:
Not sure which one you are talking about? If it's 
> expect(IdGenerationService.generateNewId()).toMatch(/[a-z 0-9]{10}/);

then I think it's checking whether the 1st 10 digits are [a-z 0-9], this will pass even if the length of the Id will be greater then 10. I would suggest you make that one to `toMatch(/[a-z 0-9]/)` and add a seperate test for length.

Reply:
> In this test we are just testing it to be a random character. I think it would be better if we name the test should generate a random id of fixed length and the expression to match would be /^[a-z 0-9]{10}$/. Let me know what you think

Sounds good to me! :)

Reply:
@DubeySandeep done